delocate/tests/test_wheelies.py

```diff
@@ -439,3 +440,27 @@ def test_fix_namespace() -> None:
         }
 
         assert delocate_wheel(NAMESPACE_WHEEL, "out.whl") == stray_libs
+
+
+def test_source_date_epoch():
+    with InTemporaryDirectory():
+        zip2dir(PURE_WHEEL, "package")
+        sde_orig = os.environ.get("SOURCE_DATE_EPOCH")
+        try:
+            for date_time, sde in (
+                ((1980, 1, 1, 0, 0, 0), 42),
+                ((1980, 1, 1, 0, 0, 0), 315532800),
+                ((1980, 1, 1, 0, 0, 2), 315532802),
+                ((2020, 2, 2, 0, 0, 0), 1580601600),
+            ):
+                os.environ["SOURCE_DATE_EPOCH"] = str(sde)
+                dir2zip("package", "package.zip")
+                with zipfile.ZipFile("package.zip", "r") as zip:
+                    for name in zip.namelist():
+                        member = zip.getinfo(name)
+                        assert_equal(member.date_time, date_time)
+        finally:
+            if sde_orig is None:
+                del os.environ["SOURCE_DATE_EPOCH"]
+            else:
+                os.environ["SOURCE_DATE_EPOCH"] = sde_orig
```

Comment:
The environment variable handing is all over the place. Would you mind rewriting this to use [contextlib.contextmanager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager)?